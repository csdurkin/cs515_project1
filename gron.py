import sys
import json

def flatten(input_object, input_path = '', input_result = None):

    if input_result is None: 
        input_result = {}

    if isinstance(input_object, dict):
        for key, value in input_object.items():
            update_path = f'{input_path}.{key}' if input_path else key
            flatten(value, update_path, input_result)

    elif isinstance(input_object, list):
        for index, value in enumerate(input_object):
            update_path = f'{input_path}[{index}]'
            flatten(value, update_path, input_result)

    else:
        input_result[input_path] = input_object

    return input_result

def gron(json_object, base_object):

    flattened_json_object = flatten(json_object)

    for key, value in flattened_json_object.items():
        print(f'{base_object}.{key} = "{value}"')


if __name__ == '__main__':
    
    try:
    
        if len(sys.argv) == 2:
            
            file_name = sys.argv[1]

            with open(file_name, 'r') as file:
                json_object = json.load(file)
            
            gron(json_object, 'json')

        elif len(sys.argv) == 4:
            
            base_object = sys.argv[2]
            file_name = sys.argv[-1]

            with open(file_name, 'r') as file:
                json_object = json.load(file)
            
            gron(json_object, base_object)

        else: 
            
            json_object = json.load(sys.stdin)
            gron(json_object)

        sys.exit(0)

    except FileNotFoundError:
        print(f"FileNotFoundError: The file '{file_name}' was not found.")
        sys.exit(1)

    except json.JSONDecodeError:
        print(f'JSONDecodeError: JSON not available in {file_name}.')
        sys.exit(1)

    except Exception as e:
        print(f'Error: {e}')
        sys.exit(1)   

    

    
