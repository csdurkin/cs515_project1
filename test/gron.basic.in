{
  "name": "Connor Durkin",
  "age": 31,
  "city": "Jersey City",
  "is_student": true
}
===
{
  "name": "Connor Durkin",
  "age": 31,
  "Address": {
    "City": "Jersey City",
    "State": "New Jersey",
    "Zip": 7307
  },
  "isStudent": true
}
===
{
  "Connor Durkin": ["31", "Jersey City, NJ, 07307", "Student"],
  "Jane Doe": ["25", "Hoboken, NJ, 07030", "Student"]
}
===
{
  "name": null,
  "address": null,
  "isStudent": null
}
===
{}

