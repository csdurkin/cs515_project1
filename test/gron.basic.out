json.name = "Connor Durkin"
json.age = "31"
json.city = "Jersey City"
json.is_student = "True"

json.name = "Connor Durkin"
json.age = "31"
json.Address.City = "Jersey City"
json.Address.State = "New Jersey"
json.Address.Zip = "7307"
json.isStudent = "True"

json.Connor Durkin[0] = "31"
json.Connor Durkin[1] = "Jersey City, NJ, 07307"
json.Connor Durkin[2] = "Student"
json.Jane Doe[0] = "25"
json.Jane Doe[1] = "Hoboken, NJ, 07030"
json.Jane Doe[2] = "Student"

json.name = "None"
json.address = "None"
json.isStudent = "None"

json: Empty JSON
